### DISCUSSION TITLE
add Chat Feature

### DISCUSSION THREAD

**Discussion by DraconicNEO** [Docile-Alligator/Infinity-For-Reddit#1357] [Ideas]
It would be great if there was access to the Chat feature in Infinity. I'm sure more people would switch to Infinity if there was access to the chat feature. Currently it's the only big reason to keep using the Stock app.

Edit: Just noticed that this was mentioned in #198 although it doesn't seem like that one went anywhere.

**Comment by reeseovine** [comment]
Afaik there's no API for chat, so it can't be natively implemented. It would be possible to wrap the website inside a section of the app but at that point it would probably just be easier to use a web browser. It may be worth keeping this issue open in case it does end up being added to the API.

**Comment by Docile-Alligator** [comment]
Hi Reddit chat is only available in the official app. There is no public API for this.

**Comment by DraconicNEO** [comment]
> 
> 
> Hi Reddit chat is only available in the official app. There is no public API for this.

How is it available in the browser?

**Comment by Elaborendum** [comment]
Wouldn't it be possible to open the chat in webview? Kind of clunky but it could at least ease some difficulties.

**Reply by Wladefant** [reply]
Hmmm. Converted it to a discussion.

**Reply by DraconicNEO** [reply]
That could work too, I had the thought to run the web version in some kind of background capacity and then scrape the data to put in our own ui (something similar to how Newpipe did it) though I'm not sure how easy that would be or if it would even work at all.

**Reply by rom4ster** [reply]
The chat is based off of matrix. So in theory reverse engineering should be possible.

**Comment by nesper8** [comment]
+1 for reddit chat

**Comment by DraconicNEO** [comment]
That seems promising, unfortunately after the events of last year I'm doubtful for any major new developments to do with this app anymore what with the new API fees and restrictions, even though he said he would keep it going by making it subscription-only I don't really feel the outlook on it is that great. 

Also because of those events it's likely he isn't going to do something not by the book so to speak since that might burn the bridge with Reddit.

If it was going to be done it would have to be via a fork of Infinity, rather than with this.
